### PR TITLE
[GH-170] Allow to overwrite pre join page inside Mattermost

### LINKED ISSUE #170: Allow to overwrite pre join page inside Mattermost
With Jitsi, you can add a "Pre join page", very useful to let people decide about the webcam/microphone enabled or not.

If you enable this in Jitsi, then the Mattermost small overlay window is too small to be able to read it. But I think an easy way would be to add a checkbox in the plugin to overload the Jitsi server config, like this:

```
const options = {
....
configOverwrite: { prejoinPageEnabled: false }
...
```

This way, people on Mattermost won't have this pre join (if they checked the box), but external people will.

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
### Summary

Added radio buttons in system console settings to disable/enable the pre-join page in Jitsi meet when opened in minimized view.

1. Added field in plugin.json file to add "Show pre-join page" radio buttons.
2. OverWrite the Jitsi server config to disable/enable pre-join page.
```
configOverwrite: {
    // Disable the pre-join page
    prejoinPageEnabled: this.props.meetingEmbedded && this.props.showPrejoinPage
}
```
3. When the pre-join page is enabled in minimized view, vertical height of the window is increased to avoid croping of the user's profile image.
4. Fixed govet fieldalignment errors.

### Screenshots

1. When pre-join page is disabled
Screenshot from 2022-07-01 15-23-25

2. When pre-join page is enabled
with_bigger_window(updated behaviour)

### Ticket
Fixes #170

### CODE REVIEW

File: webapp/src/components/conference/conference.tsx [hunk 1]
```diff
@@ -232,7 +241,7 @@ export default class Conference extends React.PureComponent<Props, State> {
 
         return (
             <div style={style.buttons}>
-                {this.state.minimized && this.state.position === POSITION_TOP &&
+                {!this.props.showPrejoinPage && this.state.minimized && this.state.position === POSITION_TOP &&
```
[the remark this answers is not in the document]
Comment (mkdbns) [reply webapp/src/components/conference/conference.tsx:244]: @larkox , there is another outstanding Jitsi ticket to enable user-resizing of the Jitsi window that the product team was going to review/prioritize. For the time being, Katie agreed that we should provide a taller window so the pre-join page isn't clipped. For users that disable the prejoin, we decided to leave the window as-is since it isn't clipped.

**Review by larkox — APPROVED** [review]

**Comment by raghavaggarwal2308** [comment]
@mickmister Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@mickmister Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@mickmister Gentle reminder for review

File: server/api.go [hunk 2]
```diff
@@ -22,8 +22,8 @@ var externalAPICacheMutex sync.Mutex
 
 type StartMeetingRequest struct {
 	ChannelID string `json:"channel_id"`
-	Personal  bool   `json:"personal"`
 	Topic     string `json:"topic"`
+	Personal  bool   `json:"personal"`
```
Comment (mickmister) [inline server/api.go:26]: Can we revert the changes in this file to keep the git history appropriate?

File: server/command_test.go [hunk 3]
```diff
@@ -69,10 +69,10 @@ func TestCommandSettings(t *testing.T) {
 	}
 
 	tests := []struct {
+		newConfig *UserConfig
```
Comment (mickmister) [inline server/command_test.go:72]: Same here to remove this change to keep git history appropriate

**Review by mickmister — APPROVED** [review]
LGTM :+1: Just some non-blocking questions/suggestions

**Comment by raghavaggarwal2308** [comment]
@mickmister Gentle reminder for re-review

**Comment by raghavaggarwal2308** [comment]
@mickmister Gentle reminder for re-review

**Comment by mickmister** [comment]
@raghavaggarwal2308 It doesn't look like there have been any changes since I approved the PR. What would you like me to review?

**Comment by raghavaggarwal2308** [comment]
@mickmister I meant the review of the answers to the questions you asked when you last reviewed.

**Comment by mickmister** [comment]
@raghavaggarwal2308 I don't see answers to the questions either though

**Comment by raghavaggarwal2308** [comment]
@mickmister Sorry about that, you should see them now

**Comment by mickmister** [comment]
Ok looks good. For future reference, mentioning me in the comments is good enough if you want to continue discussion on those individual threads. No need to ping with re-review requests if I have "approved" the PR. Hopefully that makes sense.

If the PR is only waiting on QA review, the person you want to reach out to is @DHaussermann

**Comment by mickmister** [comment]
@raghavaggarwal2308 Sorry if I seem aggressive in my previous comment. It's just a little frustrating to see this sit here like this. I'm sure it's more frustrating for you as well, sorry about this.

**Comment by raghavaggarwal2308** [comment]
@mickmister No issues, I understand :)

File: plugin.json [hunk 4]
```diff
@@ -34,6 +34,12 @@
                 "type": "bool",
                 "help_text": "(Experimental) When true, Jitsi video is embedded as a floating window inside Mattermost by default. Users can override this setting with '/jitsi settings'."
             },
+            {
+                "key": "JitsiPrejoinPage",
+                "display_name": "Show pre-join page:",
+                "type": "bool",
+                "help_text": "When false, pre-join page will not be displayed when Jitsi is embedded inside Mattermost."
```
[the remark this answers is not in the document]
Comment (mickmister) [reply plugin.json:41]: I would think that this would be a per-user setting

[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply plugin.json:41]: @mickmister We can create a per-user setting for this. What should be the UI for this, we can have a slash command `/jitsi settings` and it can post a slack attachment with buttons "Yes" and "No" or is there some other approach in your mind?

[the remark this answers is not in the document]
Comment (mickmister) [reply plugin.json:41]: `/jitsi settings pre-join true/false` should work fine

[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply plugin.json:41]: @mickmister Added the slash command

**Comment by raghavaggarwal2308** [comment]
@DHaussermann Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@DHaussermann Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@DHaussermann Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@DHaussermann Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@DHaussermann Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@DHaussermann Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@DHaussermann Gentle reminder for review

**Comment by raghavaggarwal2308** [comment]
@DHaussermann Gentle reminder for review

**Review by AayushChaudhary0001 — APPROVED** [review]
The above PR has been tested for the following features:-
- Added a config for Show pre-join page.
- Increased height when the above config is enabled.
- Height decreases when the above config is disabled.

The PR was working fine for all the above scenarios, LGTM. Approved.
